### PR TITLE
lib/connections: Limit rate at which we print warnings about version mismatch

### PR DESCRIPTION
**Pull request by calmh**
Relates to #3287. Tested manually.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as 6f5ca53f9967b543ba418f055074497b316b9015. Thanks, @calmh!

File: lib/connections/service.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ var (
 	listeners = make(map[string]listenerFactory, 0)
 )
 
+const perDeviceWarningRate = 1.0 / (15 * 60) // Once per 15 minutes
```
Comment (canton7) [inline lib/connections/service.go]: Suggest we should do this a bit more frequently? A version mis-match is almost never expected in a correct set-up: if it occurs, the user's going to be trying to find out why their devices won't connect. If the first message gets lost in some noise, we don't want them searching around for 15 mins before it pops up again, I would have thought..?

Comment (calmh) [inline lib/connections/service.go]: At least for the GUI, we stash the last five warnings and they're quite infrequent. So it should be available directly in the GUI when opened if it's happened at all recently

**Pull request closed without merging**